echo-notifications/src/test/groovy/com/netflix/spinnaker/echo/notification/AbstractEventNotificationAgentSpec.groovy

```diff
@@ -0,0 +1,139 @@
+    // notifications ON, succeeded pipeline and matching config
+    new Event(details: [type: "orca:pipeline:complete"],
+      content: [execution: [
+        id: "1",
+        name: "foo-pipeline",
+        status: 'SUCCEEDED',
+        notifications: [[type: "fake", when: "pipeline.complete"]]
+      ]])                                                                                   || 1
+    // notifications ON, succeeded pipeline and non-matching config
+    new Event(details: [type: "orca:pipeline:complete"],
+      content: [execution: [
+        id: "1",
+        name: "foo-pipeline",
+        status: 'SUCCEEDED',
+        notifications: [[type: "fake", when: "pipeline.failed"]]
+      ]])                                                                                   || 0
+    // notifications ON, failed pipeline and matching config
+    new Event(details: [type: "orca:pipeline:failed"],
+      content: [execution: [
+        id: "1",
+        name: "foo-pipeline",
+        status: 'TERMINAL',
+        notifications: [[type: "fake", when: "pipeline.failed"]]
+      ]])                                                                                   || 1
+    // notifications ON, failed pipeline and non-matching config
+    new Event(details: [type: "orca:pipeline:failed"],
+      content: [execution: [
+        id: "1",
+        name: "foo-pipeline",
+        status: 'TERMINAL',
+        notifications: [[type: "fake", when: "pipeline.complete"]]
+      ]])                                                                                   || 0
+    // notifications ON, cancelled pipeline (should skip notifications)
+    // note: this case is a bit convoluted as the event type is still set to "failed" by orca for cancelled pipelines
+    new Event(details: [type: "orca:pipeline:failed"],
+      content: [execution: [
+        id: "1",
+        name: "foo-pipeline",
+        status: 'CANCELED',
+        notifications: [[type: "fake", when: "pipeline.failed"]]
+      ]])                                                                                   || 0
+    // notifications ON, another check for cancelled pipeline (should skip notifications)
+    // note: this case is a bit convoluted as the event type is still set to "failed" by orca for cancelled pipelines
+    new Event(details: [type: "orca:pipeline:failed"],
+      content: [execution: [
+        id: "1",
+        name: "foo-pipeline",
+        status: 'WHATEVER',
+        canceled: true,
+        notifications: [[type: "fake", when: "pipeline.failed"]]
+      ]])                                                                                   || 0
+    // notifications ON, another check for cancelled pipeline (should skip notifications)
+    // note: this case is a bit convoluted as the event type is still set to "failed" by orca for cancelled pipelines
+    new Event(details: [type: "orca:pipeline:failed"],
```

Comment:
I think this case is the same as the one above?

Reply:
Indeed. Copy/paste mistake. Will fix.

Reply:
Indeed. Copy/paste mistake. Will fix.